pkg/prometheus/promcfg.go

```diff
@@ -2142,6 +2163,46 @@ func (cg *ConfigGenerator) generateK8SSDConfig(
 	}
 }
 
+func generateRoleSelectorConfig(k8sSDConfig yaml.MapSlice, roles []string, selector *metav1.LabelSelector, cg *ConfigGenerator) yaml.MapSlice {
+	selectors := []yaml.MapSlice{}
+	for _, role := range roles {
+		yml := yaml.MapSlice{
+			{
+				Key:   "role",
+				Value: role,
+			},
+		}
+
+		labelValues := []string{}
+		if len(selector.MatchLabels) > 0 {
+			labelValues = append(labelValues, labels.FormatLabels(selector.MatchLabels))
+		}
+
```

Comment:
(suggestion) We can create a labels.Selector object so we call labelSelector.Add(requirement) after.
```suggestion
		labelSelector := labels.SelectorFromValidatedSet(labels.Set(selector.MatchLabels))

```

Reply:
@simonpasquier you're a genius